packages/loot-core/src/types/models/schedule.d.ts

```diff
@@ -31,3 +31,25 @@ export interface ScheduleEntity {
   _conditions: unknown;
   _actions: unknown;
 }
+
+export type DiscoverScheduleEntity = {
+  id: ScheduleEntity['id'];
+  account: AccountEntity['id'];
+  payee: PayeeEntity['id'];
+  date: ScheduleEntity['_date'];
+  amount: ScheduleEntity['_amount'];
+  _conditions: [
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏this should use enum type instead of an array with specific values (`_conditions: Array<A | B | C>`)

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoQM5mQbwFDKHIBucANgK4QBcyA2gUUzsgPYAOtA5MBlwNzIYwCGQAm3RAlYVwA5AF8ANIyaEWHbr3nDRE5F3ZwAnhAjzlq5AF08CvHmkgsyMXDBxa6FwF5cV0koaeitmNk4DbSUhEXFuI1NzRStbBSA

Reply:
I think, it should be an array of values instead, because these values are hard-coded in the same order in the return of `find-schedules` and if we change to enum, we get error in there that the `source has 4 elements but target allows only 1`

<img width="1309" alt="Screenshot 2023-11-22 at 1 07 46 PM" src="https://github.com/actualbudget/actual/assets/62111075/7e5c2f67-2e45-498a-b4b5-c13e1475a250">

Reply:
Currently it's 4 objects, but this might change to more or less in the future. We shouldn't tie the typings to the actual business logic.

Eventually I think we'll even get rid of this `Conditions` typing and instead use a generic one across the entire product. But it'll take us some time to get there..  :)

Reply:
> shouldn't tie the typings to the actual business logic

Agree !!, however, doing it in `(_conditions: Array<A | B | C>)` is not working here, what other ways would you suggest here 🤔  ? or should we keep it as  `Array<A | B | C>` for now to change it in future ?

Reply:
Here you go: https://github.com/actualbudget/actual/pull/1946/files#r1404613308